src/modelmanager.cpp

```diff
@@ -661,15 +661,16 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
     }
 
     auto fs = ModelManager::getFilesystem(config.getBasePath());
-    std::vector<model_version_t> requestedVersions;
-    auto blocking_status = readAvailableVersions(fs, config.getBasePath(), requestedVersions);
+    std::vector<model_version_t> availableVersions;
+    Status blocking_status = readAvailableVersions(fs, config.getBasePath(), availableVersions);
     if (!blocking_status.ok()) {
         return blocking_status;
     }
-    requestedVersions = config.getModelVersionPolicy()->filter(requestedVersions);
+    auto requestedVersions = config.getModelVersionPolicy()->filter(availableVersions);
     std::shared_ptr<model_versions_t> versionsToStart;
     std::shared_ptr<model_versions_t> versionsToReload;
     std::shared_ptr<model_versions_t> versionsToRetire;
+    std::shared_ptr<model_versions_t> versionsFailed = std::make_shared<model_versions_t>(0);
```

Comment:
If intention was to create empty vector, then you do not need to pass `0`, just call:  
`std::make_shared<model_versions_t>()`